grpc-sys/build.rs

```diff
@@ -12,14 +12,32 @@
 // limitations under the License.
 
 extern crate gcc;
+extern crate cmake;
+
+fn build_grpc() {
+    let dst = cmake::Config::new("grpc")
+        .build_target("grpc")
+        .build();
+
+    println!("cargo:rustc-link-search=native={}/build", dst.display());
+    println!("cargo:rustc-link-search=native={}/build/third_party/zlib", dst.display());
+
+    println!("cargo:rustc-link-lib=static=z");
+    println!("cargo:rustc-link-lib=static=gpr");
+    println!("cargo:rustc-link-lib=static=grpc");
+}
+
 
 fn main() {
-    // TODO: support static link
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=grpc/");
+
+    build_grpc();
```

Comment:
rustc-link-lib=static means static link.

I'll figure out how to init submodules automatically.

grpc supports 2 kinds of build, cmake and bazel.